sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala

```diff
@@ -589,6 +602,7 @@ class SQLAppStatusListenerSuite extends SparkFunSuite with SharedSQLContext with
       time))
     assert(statusStore.executionsCount === 2)
     assert(statusStore.execution(2) === None)
+    store.close()
```

Comment:
I would prefer closing the old `kvstore` right before the line:
```
    val store = new ElementTrackingStore(new InMemoryStore, conf)
```

And modify the line to:
```
    kvstore = new ElementTrackingStore(new InMemoryStore, conf)
```

I know it is not a big leak but if the test fails meanwhile (because of assertion) there would no `close` called. 
On the other hand a `try` . and `finally` is can be saved this way.

Reply:
Hi @attilapiros , Thanks for the comment, I combined the both kvstore and it will create in the before block.

"retainedExecution" configuration is needed for only the "listener". Similar test is there in the "AppStatusListenerSuite", where store doesn't use listener specific conf.
https://github.com/apache/spark/blob/d3813d8b210d127ea278015ef27ead9348365787/core/src/test/scala/org/apache/spark/status/AppStatusListenerSuite.scala#L1297-L1303